online/src/test/scala/ai/chronon/online/FetcherCacheTest.scala

```diff
@@ -0,0 +1,334 @@
+package ai.chronon.online
+
+import ai.chronon.aggregator.windowing.FinalBatchIr
+import ai.chronon.api.Extensions.GroupByOps
+import ai.chronon.api.GroupBy
+import ai.chronon.online.FetcherBase._
+import ai.chronon.online.Fetcher.Request
+import ai.chronon.online.FetcherCache.{BatchIrCache, BatchResponses, CachedMapBatchResponse}
+import ai.chronon.online.KVStore.TimedValue
+import ai.chronon.online.Metrics.Context
+import org.junit.Assert.{assertArrayEquals, assertEquals, assertNull, fail}
+import org.junit.Test
+import org.mockito.Mockito._
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito
+import org.mockito.stubbing.Stubber
+import org.scalatestplus.mockito.MockitoSugar
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+trait MockitoHelper extends MockitoSugar {
+  // We override doReturn to fix a known Java/Scala interoperability issue. Without this fix, we see "doReturn:
+  // ambiguous reference to overloaded definition" errors. An alternative would be to use the 'mockito-scala' library.
+  def doReturn(toBeReturned: Any): Stubber = {
+    Mockito.doReturn(toBeReturned, Nil: _*)
+  }
+}
+
+class FetcherCacheTest extends MockitoHelper {
+  class TestableFetcherCache(cache: Option[BatchIrCache]) extends FetcherCache {
+    override val maybeBatchIrCache: Option[BatchIrCache] = cache
+  }
+  val batchIrCacheMaximumSize = 50
+
+  @Test
+  def test_BatchIrCache_CorrectlyCachesBatchIrs(): Unit = {
```

Comment:
sorry for being pedant, the code base uses camel cases only for the naming convention. Could you rename them to be camel case? Thanks!